#! /usr/bin/env python
PACKAGE='lane_extract'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name		      Type		  Reconfiguration level 			      Description			    Default	  Min 	 Max
gen.add("h_center",   int_t,  SensorLevels.RECONFIGURE_RUNNING, "Hue threshold center",   30,    0,     180)
gen.add("h_width",   int_t,   SensorLevels.RECONFIGURE_RUNNING, "Hue threshold width",    10,    1,     90)
gen.add("s_thres",   int_t,   SensorLevels.RECONFIGURE_RUNNING, "Saturation threshold",   200,    0,    255)
gen.add("v_thres",   int_t,   SensorLevels.RECONFIGURE_RUNNING, "Value threshold",        200,    0,    255)
gen.add("vote",       int_t,  SensorLevels.RECONFIGURE_RUNNING, "vote needed",            200,    0,    500)
gen.add("min_length", int_t,  SensorLevels.RECONFIGURE_RUNNING, "min length",             150,    0,    500)
gen.add("max_gap",    int_t,  SensorLevels.RECONFIGURE_RUNNING, "max gap",                10,    0,     50)
gen.add("method",    int_t,  SensorLevels.RECONFIGURE_RUNNING, "Template match method",   0,    0,     6)

exit(gen.generate(PACKAGE, PACKAGE, "lane_extract"))
